### PR TITLE
Generate coin.TRANSFER events for gas, coinbase, crosschain

### PR DESCRIPTION
**Pull request by sirlensalot**
With this change all changes to `coin` accounts will be visible via `coin.TRANSFER` events. This is also the model for a new KIP to require `TRANSFER` events for burn and create by using an empty account `""` to indicate:

- Gas payment: transfer from `sender` to miner
- Coinbase: transfer from `""` to miner
- Crosschain step 1: transfer from `sender` to `""`
- Crosschain step 2: transfer from `""` to `receiver`

TODO investigate if allocation release can be detected for generating events.

FORKING CHANGE.

### CODE REVIEW

File: stack.yaml [hunk 1]
```diff
@@ -61,11 +61,10 @@ extra-deps:
 
   # --- Custom Pins --- #
   - github: kadena-io/pact
-    commit: a9c9b92e313a680574479ffda4700533f19d16d5
+    commit: ccce78004d567966f5cfff131287fbdb3bf63121
```
Comment (larskuhtz) [inline stack.yaml:64]: This pin must also be updated in `cabal.project` and `project.nix`.

File: src/Chainweb/Version.hs [hunk 2]
```diff
@@ -858,6 +859,14 @@ enableSPVBridge Development = (>= 130)
 enableSPVBridge (FastTimedCPM g) = const $ g == pairChainGraph || g == petersonChainGraph
 enableSPVBridge _ = const True
 
+-- | Generate events for gas, coinbase, crosschain
+pactGenerateEvents :: ChainwebVersion -> BlockHeight -> Bool
+pactGenerateEvents Mainnet01 = (>= 1_600_000) -- 2021-05-07T14:14:46
+pactGenerateEvents Testnet04 = (>= 1_140_000) -- 2021-05-06T13:47:27
```
Comment (larskuhtz) [inline src/Chainweb/Version.hs:865]: 👍 Looks good. These there the estimates that I get base on current block heights:

mainnet: 2021-05-07 14:37:06
testnet: 2021-05-06 13:54:06

**Review by larskuhtz — CHANGES_REQUESTED** [review]
The changes look good to me. I would have expected that some golden test results would need to be updated. Also, are there any unit tests for events? If not, it might be good to add those.

**Comment by sirlensalot** [comment]
We will do this with #1218 instead.

**Pull request closed without merging**